cmake/IgnCodeCoverage.cmake

```
@@ -120,6 +120,13 @@ FUNCTION(ign_setup_target_for_coverage)
     MESSAGE(FATAL_ERROR "genhtml not found! Aborting...")
   ENDIF() # NOT GENHTML_PATH
 
+  # Read ignore file list
+  if (EXISTS "${PROJECT_BINARY_DIR}/coverage.ignore")
+    file (STRINGS "${PROJECT_BINARY_DIR}/coverage.ignore" IGNORE_LIST_RAW)
+    string(REGEX REPLACE "([^;]+)(;?)" "'${PROJECT_SOURCE_DIR}/\\1'\\2" IGNORE_LIST "${IGNORE_LIST_RAW}")
```

What does `\\1'\\2` for the `<replacement_expression>` do?

---

We want to use regex to change `"a.cc;b.cc"` to `"${PROJECT_SOURCE_DIR}/a.cc;${PROJECT_SOURCE_DIR}/b.cc"`. `\\1` is the matched pattern in the first pair of `()` and `\\2` is the second pair

---

I've tested the below suggestion and it still works the same. I don't think `(;?)` and `\\2` are needed.
```suggestion
    string(REGEX REPLACE "([^;]+)" "'${PROJECT_SOURCE_DIR}/\\1'" IGNORE_LIST "${IGNORE_LIST_RAW}")
```